packages/builders/src/interactions/commands/chatInput/mixins/SharedSubcommands.ts

```diff
@@ -0,0 +1,53 @@
+import {
+	ChatInputCommandSubcommandBuilder,
+	ChatInputCommandSubcommandGroupBuilder,
+} from '../ChatInputCommandSubcommands.js';
+
+export interface SharedChatInputCommandSubcommandsData {
+	options?: (ChatInputCommandSubcommandBuilder | ChatInputCommandSubcommandGroupBuilder)[];
+}
+
+/**
+ * This mixin holds symbols that can be shared in chat input subcommands.
+ *
+ * @typeParam TypeAfterAddingSubcommands - The type this class should return after adding a subcommand or subcommand group.
+ */
+export class SharedChatInputCommandSubcommands {
+	protected declare readonly data: SharedChatInputCommandSubcommandsData;
+
+	/**
+	 * Adds a new subcommand group to this command.
+	 *
+	 * @param input - A function that returns a subcommand group builder or an already built builder
+	 */
+	public addSubcommandGroup(
```

Comment:
Currently, this mixin is only being used in `ChatInputCommand`.
I think we could move this method there directly, and then we can use this mixin in `ChatInputCommandSubcommands` too.

Also, should this support adding multiple options at once?

Reply:
Hard-no towards adding multiple options in. Also this should move to the base builder bc its the only place that has it. Subcommands can't have groups